lib/travis/api/v3/renderer.rb

```diff
@@ -10,13 +10,22 @@ def clear(**args)
       args.select { |key, value| !value.nil? }
     end
 
-    def href(type, script_name: nil, **args)
-      expander     = EXPANDER_CACHE[[type, script_name, args.keys]] ||= begin
-        resource   = Routes.resources.detect { |r| r.identifier == type }
-        unprefixed = args.keys.reject { |a| a.to_s.include? ?..freeze }
-        route      = resource.route
-        route    &&= Mustermann.new(script_name, type: :identity) + route if script_name and not script_name.empty?
-        generate_expander(route, type, unprefixed)
+    def href(type, string_args = nil, script_name: nil, **args)
+      args.merge! string_args if string_args
+
+      expander      = EXPANDER_CACHE[[type, script_name, args.keys]] ||= begin
+        resource    = Routes.resources.detect { |r| r.identifier == type }
+        route       = resource.route
+        route     &&= Mustermann.new(script_name, type: :identity) + route if script_name and not script_name.empty?
+        key_mapping = {}
+        args.keys.each do |key|
+          case key.to_s
+          when /\./        then key_mapping[key.to_sym]        = key unless key.is_a? Symbol
+          when /^(.+)_id$/ then key_mapping[:"#{$1}.id"]       = key
+          else                  key_mapping[:"#{type}.#{key}"] = key
+          end
+        end
+        generate_expander(route, key_mapping)
```

Comment:
This change is to allow links generated from other attributes than id (like branch.name). See below.